edd-purchase-gravatars.php

```diff
@@ -276,13 +271,14 @@ public function gravatars( $download_id = false, $title = '' ) {
 
 			}
 
-			$i = 0;
+			$i       = 0;
+			$maximum = edd_get_option( 'edd_pg_maximum_number' );
 
 			if ( $payment_ids ) {
 				foreach ( $payment_ids as $id ) {
 
 					// EDD saves a blank option even when the control is turned off, hence the extra check
-					if ( isset( $edd_options['edd_pg_maximum_number'] ) && '' != $edd_options['edd_pg_maximum_number'] && $i == $edd_options['edd_pg_maximum_number'] ) {
+					if ( ! empty( $maximum ) && $i >= (int) $maximum ) {
 						continue;
```

Comment:
```suggestion
						break;
```

I know it's not part of your PR, but `continue` will just make it keep going, and keep skipping, all until it's done. Whereas `break` will actually end the `foreach` loop completely, which is more in line with what we want.